app/ports/models/port.py

```diff
@@ -0,0 +1,335 @@
+import json
+import os
+import subprocess
+
+from django.db import models
+
+import MacPorts.config as config
+
+
+class Category(models.Model):
+    name = models.TextField(primary_key=True)
+
+
+class PortManager(models.Manager):
+    def get_by_natural_key(self, name):
+        return self.get(name=name)
+
+
+class Port(models.Model):
+    portdir = models.CharField(max_length=100)
+    description = models.TextField(default='')
+    homepage = models.URLField(default='')
+    epoch = models.BigIntegerField(default=0)
+    platforms = models.TextField(null=True)
+    categories = models.ManyToManyField(Category, related_name='category', db_index=True)
```

Comment:
This now creates a column `category_id` which suggests a numeric value, but actually holds a text value with full category name.

The full table is
* `id` = primary key (not really needed anywhere, I assume?)
* `port_id` = foreign key to `port_id`; numeric
* `category_id` = foreign key to `category_name`; text

The last two fields look somewhat inconsistently used.